cosmwasm/packages/sgx-vm/src/wasmi/wrapper.rs

```diff
@@ -133,18 +133,20 @@ where
         // At this point we know that the ecall was successful and init_result was initialized.
         let init_result = unsafe { init_result.assume_init() };
 
-        init_result_to_result_initsuccess(init_result)
-            .map(|success| {
-                trace!(
-                    target: module_path!(),
-                    "init() returned with gas_used: {} (gas_limit: {})",
-                    success.used_gas(),
-                    self.gas_left
-                );
-                self.gas_left -= success.used_gas();
-                success
-            })
-            .map_err(Into::into)
+        let used_gas = match init_result {
+            InitResult::Success { used_gas, .. } => used_gas,
+            InitResult::Failure { used_gas, .. } => used_gas,
+        };
+
+        trace!(
+            target: module_path!(),
+            "init() returned with gas_used: {} (gas_limit: {})",
+            used_gas,
+            self.gas_left
+        );
+        self.gas_left -= used_gas;
```

Comment:
@assafmo I think the panic comes from here because now when we run out of gas it tries to subtract the gas_left below 0, but it's a unsigned integer and that panics in debug mode. (that's a good thing). I'll refactor this to either use `saturating_sub` or replace `gas_left` with `gas_used` so we can go higher. I'd still use `saturating add` but we won't lose information about how much gas was used unless someone actually uses more than `u64::MAX` gas points.
This will be very similar to what we do in `ContractInstance::use_gas` inside the enclave.

Reply:
yeah it even says in the logs:
```
 thread '<unnamed>' panicked at 'attempt to subtract with overflow', /home/runner/work/SecretNetwork/SecretNetwork/cosmwasm/packages/sgx-vm/src/wasmi/wrapper.rs:195:9
```

Reply:
yup, tests pass for 08d837e